File: 0624-maximum-distance-in-arrays/0624-maximum-distance-in-arrays.cpp
class Solution {
public:
    int maxDistance(vector<vector<int>>& arrays) {
        int a=arrays[0][0];
        int b=arrays[0].back();
        int res=0;
        for(int i=1;i<arrays.size();i++){
            res=max(res,abs(arrays[i].back()-a));
            res=max(res,abs(arrays[i][0]-b));
            a=min(a,arrays[i][0]);
            b=max(b,arrays[i].back());
        }
        return res;
    }
};

File: 0624-maximum-distance-in-arrays/0624-maximum-distance-in-arrays.py
class Solution:
    def maxDistance(self, arrays: List[List[int]]) -> int:
        a=arrays[0][0]
        b=arrays[0][-1]
        res=0
        for i in arrays[1:]:
            res=max(res,abs(i[-1]-a),abs(b-i[0]))
            b=max(i[-1],b)
            a=min(i[0],a)
        return res